blockchain/standalone/pow_test.go

```diff
@@ -393,3 +393,1676 @@ func TestCheckProofOfWork(t *testing.T) {
+			{226, 668981, 0x1c271457},
+			{227, 670303, 0x1c27b995},
+			{228, 670467, 0x1c27a36c},
+			{229, 671633, 0x1c283135},
+			{230, 676603, 0x1c2b5231},
+			{231, 680345, 0x1c2dc7f3},
+			{232, 685034, 0x1c311e3c},
+			{233, 692431, 0x1c37092d},
+			{234, 694103, 0x1c3842ad},
+			{235, 694962, 0x1c38c46a},
+			{236, 699094, 0x1c3c5e51},
+			{237, 705075, 0x1c422298},
+			{238, 709010, 0x1c461c6d},
+			{239, 710982, 0x1c4804cc},
+			{240, 713828, 0x1c4b05d6},
+			{241, 716627, 0x1c4e175f},
+			{242, 719236, 0x1c5108e0},
+			{243, 719557, 0x1c510fd5},
+			{244, 721094, 0x1c52af16},
+			{245, 724674, 0x1c572909},
+			{246, 731160, 0x1c604034},
+			{247, 731204, 0x1c5fdb43},
+			{248, 737107, 0x1c68dcc0},
+			{249, 737314, 0x1c68b4a2},
+			{250, 737578, 0x1c68a56d},
+			{251, 738726, 0x1c6a13fd},
+			{252, 745471, 0x1c75a1b8},
+			{253, 748735, 0x1c7b5d43},
+			{254, 749877, 0x1c7d0ae9},
+			{255, 750488, 0x1c7dab62},
+			{256, 754736, 0x1d0085e5},
+			{257, 755780, 0x1d008781},
+			{258, 757308, 0x1d008a34},
+			{259, 761267, 0x1d009291},
+			{260, 761700, 0x1d0092e1},
+			{261, 768268, 0x1d00a269},
+			{262, 771486, 0x1d00aa33},
+			{263, 771983, 0x1d00aabd},
+			{264, 772304, 0x1d00aacc},
+			{265, 775659, 0x1d00b362},
+			{266, 776129, 0x1d00b3e0},
+			{267, 778615, 0x1d00ba4b},
+			{268, 782911, 0x1d00c69d},
+			{269, 785122, 0x1d00ccca},
+			{270, 792634, 0x1d00e5e6},
+			{271, 806060, 0x1d00ffff},
+			{272, 813100, 0x1d00ffff},
+			{273, 820611, 0x1d00ffff},
+			{274, 821642, 0x1d00ffff},
+			{275, 823392, 0x1d00ffff},
+			{276, 824267, 0x1d00ffff},
+			{277, 825093, 0x1d00ffff},
+			{278, 826268, 0x1d00ffff},
+			{279, 826731, 0x1d00ffff},
+			{280, 833358, 0x1d00ffff},
+			{281, 833366, 0x1d00ffff},
+			{282, 835141, 0x1d00ffff},
+			{283, 843181, 0x1d00ffff},
+			{284, 846510, 0x1d00ffff},
+			{285, 848500, 0x1d00ffff},
+			{286, 855266, 0x1d00ffff},
+			{287, 857364, 0x1d00ffff},
+			{288, 857850, 0x1d00ffff},
+		},
+	}, {
+		name:          "mainnet: 300s blocks across signed 32-bit int height",
+		params:        &mockMainNetParams,
+		startDiffBits: 0x1b00a5a6,
+		startHeight:   2147483645,
+		startTime:     7157978,
+		perInstanceTests: []perInstanceTest{
+			{2147483646, 7158278, 0x1b00a5a6},
+			{2147483647, 7158578, 0x1b00a5a6},
+			{2147483648, 7158878, 0x1b00a5a6},
+		},
+	}, {
+		name: "mainnet: 300s blocks across max 64-bit signed int height and " +
+			"32-bit max signed int time",
+		params:        &mockMainNetParams,
+		startDiffBits: 0x1b00a5a6,
+		startHeight:   9223372036854775804,
+		startTime:     2147482747,
+		perInstanceTests: []perInstanceTest{
+			{9223372036854775805, 2147483047, 0x1b00a5a6},
+			{9223372036854775806, 2147483347, 0x1b00a5a6},
+			{9223372036854775807, 2147483647, 0x1b00a5a6},
+			{9223372036854775808, 2147483947, 0x1b00a5a6},
+			{9223372036854775809, 2147484247, 0x1b00a5a6},
+		},
+	}, {
+		name: "mainnet: blocks across potential internal calculation overflow" +
+			" boundaries",
```

Comment:
While debugging the tests, I noticed there was no test exercising the case where the internal `frac64` is set to its maximum amount (`0xfff`). Here is a test case to exercise that:

```go
{                                                                                                 
        name:          "mainnet: blocks that exercise max fractional part in internal calculation",  
        params:        &mockMainNetParams,                                                           
        startDiffBits: 0x1b00a5a6,                                                                   
        startHeight:   0,                                                                            
        startTime:     0,                                                                            
        perInstanceTests: []perInstanceTest{                                                         
                {1, 299, 0x1b00a5a5},                                                                
                {2, 599, 0x1b00a5a5},                                   
                {3, 899, 0x1b00a5a5},                                                                
                {4, 1199, 0x1b00a5a5},                                                               
        },                                                                                           
}
```

Reply:
Nice.  It's a good idea to add a vector for this case as well.  I will not however that these overflow tests do hit the edges where the inner addition requires all 64 bits for a correct result (all of them require the full 64 bits for the inner 16.48 addition).

Thanks for the additional test vectors!

Added.

Reply:
Nice.  It's a good idea to add a vector for this case as well.  I will not however that these overflow tests do hit the edges where the inner addition requires all 64 bits for a correct result (all of them require the full 64 bits for the inner 16.48 addition).

Thanks for the additional test vectors!

Added.